concent_api/core/tests/test_core_views.py

```diff
@@ -496,6 +510,98 @@ def test_send_should_return_http_202_if_task_to_compute_deadline_is_correct(self
 
             self.assertIn(response_202.status_code, [200, 202])
 
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_task_to_compute_without_public_key_should_return_http_400(self):
```

Comment:
There is some code duplication with the function below; with some small refactoring you can get sth like this:
```
@freeze_time("2017-11-17 10:00:00")
    @pytest.mark.parametrize('attribute', [
        'provider_public_key',
        'requestor_public_key',
    ])
    def test_send_task_to_compute_without_public_key_should_return_http_400(self, attribute):
        assert StoredMessage.objects.count() == 0
        setattr(self.task_to_compute, attribute, None)

        response = self.send_force_report_computed_task(self.task_to_compute)

        self.assert_bad_request(response)
```

Reply:
Not exactly, first test is setting attribute to None, while second is cutting its length.

Reply:
second test could be like this:
```
@freeze_time("2017-11-17 10:00:00")
@pytest.mark.parametrize('attribute', [
        'provider_public_key',
        'requestor_public_key',
])
def test_send_task_to_compute_with_public_key_with_wrong_length_should_return_http_400(self, attribute):
        assert StoredMessage.objects.count() == 0
        setattr(
                self.task_to_compute,
                attribute,
                getattr(self.task_to_compute, attribute)[:-1]
            )
        response = self.send_force_report_computed_task(self.task_to_compute)

        self.assert_bad_request(response)
```